DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseSetupControllerTests.cs

```diff
@@ -151,22 +219,446 @@ public void Index_with_CLEAR_filterBy_and_new_filter_query_parameter_sets_new_co
 
         [Test]
         public void Index_with_no_filtering_should_default_to_Active_courses()
+        {
+            // When
+            var result = controller.Index();
+
+            // Then
+            result.As<ViewResult>().Model.As<CourseSetupViewModel>().FilterBy.Should()
+                .Be("Status|Active|true");
+        }
+
+        [Test]
+        public void AddCourseNew_sets_new_temp_data()
+        {
+            // When
+            var result = controller.AddCourseNew();
+
+            // Then
+            controller.TempData.Peek<AddNewCentreCourseData>().Should().NotBeNull();
+            result.Should().BeRedirectToActionResult().WithActionName("SelectCourse");
+        }
+
+        [Test]
+        public void SelectCourse_post_updates_temp_data_and_redirects()
+        {
+            var applicationOptionsSelectList = new List<SelectListItem> { new SelectListItem("Test Name", "1") };
+            var model = new SelectCourseViewModel(application.ApplicationId, applicationOptionsSelectList);
+            SetAddNewCentreCourseTempData();

```

Comment:
I'm not sure on the need for this line. The actual controller should set this data.
I think manipulating the temp data here may void the benefits of the assertions below.